src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -0,0 +1,216 @@
+using Standard;
+using System.Windows.Appearance;
+using System.Windows.Media;
+using Microsoft.Win32;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows.Interop;
+
+namespace System.Windows;
+
+internal static class ThemeManager
+{
+
+    #region Constructor
+
+    static ThemeManager()
+    {
+        _cachedThemeDictionaryUris = new List<Uri>();
+
+        if (Application.Current != null)
+        {
+            foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
+            {
+                if (mergedDictionary.Source != null && mergedDictionary.Source.ToString().EndsWith("FluentWindows.xaml"))
+                {
+                    _isFluentWindowsThemeEnabled = true;
+                    break;
+                }
+            }
+        }
+
+        if(_isFluentWindowsThemeEnabled)
+        {
+            _currentApplicationTheme = GetSystemTheme();
+            DwmColorization.UpdateAccentColors();
+        }
+    }
+
+    #endregion
+
+    #region Internal Methods
+
+    internal static void ApplySystemTheme(bool forceUpdate = false)
+    {
+        string systemTheme = GetSystemTheme();
+        bool useLightMode = GetUseLightTheme();
```

Comment:
It is confusing that both these method says Theme but you store one as "theme" and the other one as "mode".  GetUseLightTheme is also not a great name because it has two verbs.  I am almost tempted to suggest turning these into properties instead, like `SystemThemeName` and `UseLightTheme` or `IsThemeLight` or similar.

Reply:
GetUseLightTheme comes from the registry key name "AppsUseLightTheme". I didn't want to use a property named IsThemeLight as it gives the sense that this is the application's current color mode.

Reply:
or use one of the enums, didn't we have an enum for dark/light?

Reply:
GetSystemColor ? or as a property SystemThemeColor ?

Regarding that enum ApplicationTheme, I am thinking or removing that. Currently it is not being used and it is not accurate as it combines color and theme together